q2galaxy/core/drivers/builtins.py

```diff
@@ -59,15 +60,18 @@ def _import_get_args(inputs):
             raise ValueError(f"Unknown instruction in JSON: {key}")
         elif key == 'import':
             # leave name as is, it's a FileFormat not a Directory Attr
-            files_to_move.append((value['data'], value['data']))
+            files_to_move.append((value['source_path'], value['source_path']))
         else:
             _, attr_name = key.split("_")
             if 'elements' in value:
                 ext = value.get('ext', '')
                 files_to_move.extend([
-                    (v['data'], v['name'] + ext) for v in value['elements']])
+                    (v['source_path'],
+                     v['staging_path'].split('.fastqsanger.gz')[0] + ext)
```

Comment:
The `.fastqsanger.gz` suffix was only added during test file upload (haven't tested this a lot .. but might be a Galaxy bug). This should usually not appear. I guess it might work all the time.

Reply:
Yeah that was something I was curious about, and this little split here feels bad and hacky, I'll be trying to figure out what exactly is going on there if I can.

Reply:
I researched this a bit. `"staging_path"` will be `"{element_identifier}.{ext}"`, where

- `element_identifier` is the name of a collection element or the dataset name
  - `"/"` characters in identifiers are replaced by `"_"`
  - for collections this is prefixed by a `"/"` separated list of collection identifiers (for nested collections there could be more than one `"/"`)
- `"ext"` is the datatype used in Galaxy. This will always be set, i.e. I was wrong that this only happens in tool tests. 
  The available datatypes used in Galaxy (https://github.com/galaxyproject/galaxy/blob/dev/lib/galaxy/config/sample/datatypes_conf.xml.sample)

I'm currently working on a PR that would add `"element_identifier"`, `"ext"` and `"name"` (which is the dataset name .. which is seldomly useful in tools). Until this is done we have to resort to hack solutions (I guess), I guess most secure would be to remove only the longest suffix matching a Galaxy datatype.